### PR TITLE
Enable Style/InvertibleUnlessCondition cop

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
This is a new cop [introduced](https://github.com/rubocop/rubocop/pull/11432) in rubocop v1.44.0. It's disabled by default but i think it's enough of a readability improvement to enable. (I also added a rule to favor `if a != b` over `unless a == b`.)

It is [invalid](https://github.com/rubocop/rubocop/issues/11800) in at least one case – when comparing class hierarchies that do not share inheritance (see the disable below).

cc @issyl0

### CODE REVIEW

File: Library/Homebrew/cmd/list.rb [hunk 1]
```diff
@@ -211,8 +211,8 @@ def print_dir(root)
       elsif block_given? && yield(pn)
         puts pn
         other = "other "
-      else
-        remaining_root_files << pn unless pn.basename.to_s == ".DS_Store"
+      elsif pn.basename.to_s != ".DS_Store"
+        remaining_root_files << pn
```
Comment (dduugg) [inline Library/Homebrew/cmd/list.rb:215]: This is an especially nice fix, imo

**Comment by dduugg** [comment]
Currently failing CI due to upstream violations, will hold off on fixing until initial feedback is in

File: Library/Homebrew/cmd/list.rb (hunk 1 above)
Comment (issyl0) [reply Library/Homebrew/cmd/list.rb:215]: Yes, `else` ... `unless` was weird!

**Review by issyl0 — COMMENTED** [review]
Thanks @dduugg, nice discovery. I have some (opinionated) comments about the corrections here.

File: Library/Homebrew/cask/audit.rb [hunk 2]
```diff
@@ -526,7 +526,7 @@ def audit_livecheck_version
     def audit_livecheck_min_os
       return unless online?
       return unless cask.livecheckable?
-      return unless cask.livecheck.strategy == :sparkle
+      return if cask.livecheck.strategy != :sparkle
```
[the remark this answers is not in the document]
Comment (dduugg) [reply Library/Homebrew/cask/audit.rb:529]: There's no easy general fix here that i can think of. Sometimes I chain these (`return unless a && b && c`), but i generally only use `unless` when rubocop makes me.

File: Library/Homebrew/cmd/cleanup.rb [hunk 3]
```diff
@@ -57,7 +57,7 @@ def cleanup
 
     cleanup.clean!
 
-    unless cleanup.disk_cleanup_size.zero?
+    if cleanup.disk_cleanup_size.nonzero?
```
[the remark this answers is not in the document]
Comment (dduugg) [reply Library/Homebrew/cmd/cleanup.rb]: That's a reasonable objection, i'll unset the entry that does this and re-apply it.

File: Library/Homebrew/utils/analytics.rb [hunk 4]
```diff
@@ -473,7 +473,7 @@ def table_output(category, days, results, os_version: false, cask_install: false
                "#{formatted_count} | #{formatted_percent}%"
           next if index > 10
         end
-        return unless results.length > 1
+        return if results.length <= 1
```
[the remark this answers is not in the document]
Comment (dduugg) [reply Library/Homebrew/utils/analytics.rb:476]: Not so for me (mostly bc i still have to mentally convert `unless` to `if not`, which makes comparisons like this a 🤕), but we could also unset these in the config, like for the `zero?` case.

File: Library/Homebrew/cmd/cleanup.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (issyl0) [reply Library/Homebrew/cmd/cleanup.rb]: Thanks!

File: Library/Homebrew/cask/audit.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeMcQuaid) [reply Library/Homebrew/cask/audit.rb:529]: I also prefer the cop's approach here.

File: Library/Homebrew/utils/analytics.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (MikeMcQuaid) [reply Library/Homebrew/utils/analytics.rb:476]: I also prefer the way the cop handles this.

**Review by MikeMcQuaid — APPROVED** [review]
Love this cop, enabling it on personal projects too, thanks @dduugg!

**Comment by MikeMcQuaid** [comment]
As a general note: I have on my "one day" TODO list to go through literally every disabled by default cop and try to enable them all and remove the ones I disagree with. Here's a list from a work project I've done this on, if it's useful:

```
# Enable useful some rules disabled by default.
Layout/ClassStructure:
  Enabled: true
Layout/EmptyLineAfterMultilineCondition:
  Enabled: true
Layout/FirstArrayElementLineBreak:
  Enabled: true
Layout/FirstHashElementLineBreak:
  Enabled: true
Layout/FirstMethodParameterLineBreak:
  Enabled: true
Layout/HeredocArgumentClosingParenthesis:
  Enabled: true
Layout/MultilineArrayLineBreaks:
  Enabled: true
Layout/MultilineHashKeyLineBreaks:
  Enabled: true
Layout/MultilineMethodParameterLineBreaks:
  Enabled: true
Layout/SingleLineBlockChain:
  Enabled: true
Lint/HeredocMethodCallPosition:
  Enabled: true
Naming/InclusiveLanguage:
  Enabled: true
Style/AutoResourceCleanup:
  Enabled: true
Style/CollectionMethods:
  Enabled: true
Style/ImplicitRuntimeError:
  Enabled: true
Style/InvertibleUnlessCondition:
  Enabled: true
Style/IpAddresses:
  Enabled: true
Style/OptionHash:
  Enabled: true
Style/ReturnNil:
  Enabled: true
Style/StringHashKeys:
  Enabled: true
Style/Send:
  Enabled: true
Style/StringMethods:
  Enabled: true
Style/TopLevelMethodDefinition:
  Enabled: true
Style/UnlessLogicalOperators:
  Enabled: true
Style/YodaExpression:
  Enabled: true
Performance/ChainArrayAllocation:
  Enabled: true
Performance/IoReadlines:
  Enabled: true
Performance/OpenStruct:
  Enabled: true
Performance/SelectMap:
  Enabled: true
Performance/CaseWhenSplat:
  Enabled: true
Rails/DefaultScope:
  Enabled: true
Rails/EnvironmentVariableAccess:
  Enabled: true
Rails/OrderById:
  Enabled: true
Rails/PluckId:
  Enabled: true
Rails/RequireDependency:
  Enabled: true
Rails/ReversibleMigrationMethodDefinition:
  Enabled: true
Rails/SaveBang:
  Enabled: true
Rails/TableNameAssignment:
  Enabled: true
RSpec/Dialect:
  Enabled: true
RSpec/MessageExpectation:
  Enabled: true
RSpec/Pending:
  Enabled: true
Sorbet/CallbackConditionalsBinding:
  Enabled: true
Sorbet/EnforceSignatures:
  Enabled: true
Sorbet/ForbidIncludeConstLiteral:
  Enabled: true
Sorbet/ForbidSuperclassConstLiteral:
  Enabled: true
Sorbet/ForbidTUnsafe:
  Enabled: true
Sorbet/OneAncestorPerLine:
  Enabled: true
Sorbet/SingleLineRbiClassModuleDefinitions:
  Enabled: true
```

**Comment by issyl0** [comment]
@MikeMcQuaid Don't tempt me to do your TODO list items for you. It's a good idea.

File: Library/Homebrew/cask/audit.rb [hunk 5]
```diff
@@ -145,7 +145,7 @@ def audit_single_pre_postflight
         k.is_a?(Artifact::PostflightBlock) &&
           k.directives.key?(:postflight)
       end
-      return unless count > 1
+      return if count <= 1
 
       add_error "only a single postflight stanza is allowed"
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:150]:
```suggestion
      add_error "only a single postflight stanza is allowed" if count > 1
```

Comment (reitermarkus) [reply Library/Homebrew/cask/audit.rb:150]: Unless RuboCop wants a guard-clause here.

**Comment by dduugg** [comment]
> @MikeMcQuaid Don't tempt me to do your TODO list items for you. It's a good idea.

Note that per [comment](https://github.com/rubocop/rubocop/pull/11432#issuecomment-1381791030) on the rubocop PR, rubocop will be enabling fewer of these kinds of cops going forward. If you'd prefer to stay on top of these as they come in, it may be worth exploring the `EnabledByDefault` [config](https://docs.rubocop.org/rubocop/configuration.html#enabled) setting.

**Comment by MikeMcQuaid** [comment]
> Note that per [comment](https://github.com/rubocop/rubocop/pull/11432#issuecomment-1381791030) on the rubocop PR, rubocop will be enabling fewer of these kinds of cops going forward. If you'd prefer to stay on top of these as they come in, it may be worth exploring the `EnabledByDefault` [config](https://docs.rubocop.org/rubocop/configuration.html#enabled) setting.

Good idea. I knew about `NewCops: enable` (which we've enabled) but I do think `EnabledByDefault` probably makes more sense. When new cops are added and we have no violations: I think it's worth us enabling them.

**Comment by issyl0** [comment]
As long as it's not `DisabledByDefault`, I'm game for config changes. Bad news though, currently running `brew style` after setting `EnabledByDefault: true` reveals `1170 files inspected, 42077 offenses detected, 21649 offenses autocorrectable`. 💀 But that's why Mike is wise and said "only enable new cops if no offenses" (or, I guess, minimal, sensible, easily fixable ones). I'll make an issue over the weekend for determining which ones are good/bad/enforceable.

**Comment by dduugg** [comment]
> As long as it's not `DisabledByDefault`, I'm game for config changes. Bad news though, currently running `brew style` after setting `EnabledByDefault: true` reveals `1170 files inspected, 42077 offenses detected, 21649 offenses autocorrectable`. 💀 But that's why Mike is wise and said "only enable new cops if no offenses" (or, I guess, minimal, sensible, easily fixable ones). I'll make an issue over the weekend for determining which ones are good/bad/enforceable.

I'm not too surprised by that. It might be more worthwhile to attempt it by department (e.g. `Style`) and not globally.

**Pull request merged**